### PR TITLE
Use soft assay response to set multi-assay fields

### PR DESCRIPTION
**Pull request by lchoy**
The response from the soft assay endpoint is used to determine the multi-assay specific fields for relevant datasets. The soft assay response contains `is-multi-assay` for primary and derived datasets of a multi-assay.

```
{
    "_shards": {
        "failed": 0,
        "skipped": 0,
        "successful": 1,
        "total": 1
    },
    "hits": {
        "hits": [
            {
                "_id": "500af39b83bfa9eb27f78746734f3db9",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "snRNAseq (SNARE-seq2)"
                    ],
                    "assay_modality": "single",
                    "dataset_type": "RNAseq",
                    "mapped_data_types": [
                        "snRNAseq (SNARE-seq2)"
                    ],
                    "processing": "raw",
                    "uuid": "500af39b83bfa9eb27f78746734f3db9",
                    "vitessce-hints": []
                },
                "_type": "_doc"
            },
            {
                "_id": "0b98992fa198c4c4a7bdd776a970616d",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "Histology"
                    ],
                    "assay_modality": "multiple",
                    "dataset_type": "Histology",
                    "is_component": true,
                    "mapped_data_types": [
                        "Histology"
                    ],
                    "processing": "raw",
                    "uuid": "0b98992fa198c4c4a7bdd776a970616d",
                    "vitessce-hints": [
                        "unknown-assay"
                    ]
                },
                "_type": "_doc"
            },
            {
                "_id": "7b2b9029035d46c4ef8306fa82c8e58e",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "CODEX [Cytokit + SPRM]"
                    ],
                    "assay_modality": "single",
                    "dataset_type": "CODEX [Cytokit + SPRM]",
                    "mapped_data_types": [
                        "CODEX [Cytokit + SPRM]"
                    ],
                    "processing": "processed",
                    "processing_type": "hubmap",
                    "uuid": "7b2b9029035d46c4ef8306fa82c8e58e",
                    "vitessce-hints": [
                        "sprm",
                        "anndata",
                        "is_image",
                        "is_tiled"
                    ]
                },
                "_type": "_doc"
            },
            {
                "_id": "ef7b17140c217fbf8ae8810b79c0e051",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "RNAseq"
                    ],
                    "assay_modality": "multiple",
                    "dataset_type": "RNAseq",
                    "is_component": true,
                    "mapped_data_types": [
                        "RNAseq"
                    ],
                    "processing": "raw",
                    "uuid": "ef7b17140c217fbf8ae8810b79c0e051",
                    "vitessce-hints": [
                        "unknown-assay"
                    ]
                },
                "_type": "_doc"
            },
            {
                "_id": "72ec02cf1390428c1e9dc2c88928f5f5",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "Visium (no probes) [Salmon + Scanpy]"
                    ],
                    "assay_modality": "multiple",
                    "dataset_type": "Visium (no probes) [Salmon + Scanpy]",
                    "is_component": false,
                    "mapped_data_types": [
                        "Visium (no probes) [Salmon + Scanpy]"
                    ],
                    "processing": "processed",
                    "processing_type": "hubmap",
                    "uuid": "72ec02cf1390428c1e9dc2c88928f5f5",
                    "vitessce-hints": [
                        "rna",
                        "is_image",
                        "anndata",
                        "spatial"
                    ]
                },
                "_type": "_doc"
            },
            {
                "_id": "a6e6ba302926dff92c599199ea1a7d6c",
                "_index": "hm_consortium_portal",
                "_score": 1.0,
                "_source": {
                    "assay_display_name": [
                        "Visium (No probes)"
                    ],
                    "assay_modality": "multiple",
                    "dataset_type": "Visium (no probes)",
                    "is_component": false,
                    "mapped_data_types": [
                        "Visium (No probes)"
                    ],
                    "processing": "raw",
                    "uuid": "a6e6ba302926dff92c599199ea1a7d6c",
                    "vitessce-hints": []
                },
                "_type": "_doc"
            }
        ],
        "max_score": 1.0,
        "total": {
            "relation": "eq",
            "value": 6
        }
    },
    "timed_out": false,
    "took": 13
}
```

### CODE REVIEW

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py [hunk 1]
```diff
@@ -33,6 +48,58 @@ def _get_assay_details(doc, transformation_resources):
...
+        doc['multi_assay_category'] = 'component'
+        return
+
+    if assay_details.get('is-multi-assay', False):
+        doc['assay_modality'] = 'multiple'
+        creation_action = doc.get('creation_action', None)
+        if creation_action == CreationAction.CREATE_DATASET:
+            doc['multi_assay_category'] = 'primary'
+        elif creation_action == CreationAction.CENTRAL_PROCESS:
+            doc['multi_assay_category'] = 'processed'
```
Comment (john-conroy) [inline src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:77]: Feels a bit in conflict with the values in `processing`. Could this just be a `is_component` bool?

File: src/hubmap_translation/addl_index_transformations/portal/utils.py [hunk 2]
```diff
@@ -6,5 +6,6 @@
 
 
 def _log_transformation_error(doc, msg):
+    doc['transformation_errors'] = doc.get('transformation_errors', [])
```
Comment (john-conroy) [inline src/hubmap_translation/addl_index_transformations/portal/utils.py:9]: When did you run into an issue that prompted this? `transformation_errors` is set in https://github.com/hubmapconsortium/search-api/blob/2c0673cd7ff33cf525815f807ca124f132a9daff/src/hubmap_translation/addl_index_transformations/portal/__init__.py#L72.

**Review by john-conroy — APPROVED** [review]
Thanks!

File: src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py (hunk 1 above)
Comment (lchoy) [reply src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py:77]: I share your concern here, especially as it creates the potential for a conflict with `processing`. Let's change this to `is_component`.

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**